release/nightly_tests/dataset/read_parquet_benchmark.py

```diff
@@ -0,0 +1,90 @@
+import pyarrow as pa
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+from parquet_data_generator import generate_data
+
+import shutil
+import tempfile
+
+
+def read_parquet(
+    root: str,
+    parallelism: int = -1,
+    use_threads: bool = False,
+    filter=None,
+    columns=None,
+) -> Dataset:
+    return ray.data.read_parquet(
+        paths=root,
+        parallelism=parallelism,
+        use_threads=use_threads,
+        filter=filter,
+        columns=columns,
+    ).fully_executed()
+
+
+def run_read_parquet_benchmark(benchmark: Benchmark):
+    # Test with different parallelism (multi-processing for single node) and threading.
+    for parallelism in [1, 2, 4]:
+        for use_threads in [True, False]:
+            test_name = f"read-parquet-downsampled-nyc-taxi-2009-{parallelism}-{use_threads}"  # noqa: E501
+            benchmark.run(
+                test_name,
+                read_parquet,
+                root="s3://anonymous@air-example-data/ursa-labs-taxi-data/downsampled_2009_full_year_data.parquet",  # noqa: E501
+                parallelism=parallelism,
+                use_threads=use_threads,
+            )
+
+    # Test with projection and filter pushdowns.
+    # Since we have projection and filter pushdown, we can run the read on the full
+    # size of one year data fast enough on a single node.
+    test_name = "read-parquet-nyc-taxi-2018-pushdown"
+    filter_expr = (pa.dataset.field("passenger_count") <= 10) & (
+        pa.dataset.field("passenger_count") > 0
+    )
+    benchmark.run(
+        test_name,
+        read_parquet,
+        root="s3://anonymous@air-example-data/ursa-labs-taxi-data/by_year/2018",
+        columns=["passenger_count", "trip_distance"],
+        filter=filter_expr,
+    )
+
+    # Test with different number files to handle: from a few to many.
+    data_dirs = []
+    # Each test set has same total number of rows, which are distributed
+    # to different number of files.
+    total_rows = 1024 * 1024 * 8
+    for num_files in [8, 128, 1024]:
+        for compression in ["snappy", "gzip"]:
```

Comment:
thanks, we have `snappy` and `gzip`, but not `zstd`.
`gzip` is not widely used in production afaik.

Reply:
gzip is quite common. Actually Pandas doesn't support zstd (maybe it's still emerging).

Reply:
I don't have data, but this thread claims snappy and gzip are the most commonly used: https://stackoverflow.com/questions/51203639/what-are-the-compression-types-supported-in-parquet

Reply:
Cool, then let's keep `snappy` and `gzip` for now.

Reply:
Agreed with `snappy` and `gzip`, I have a fuzzy sense that those are the most popular based on user convos, but I'd love to find more concrete data at some point.

>Actually Pandas doesn't support zstd (maybe it's still emerging).

Btw @jianoaix , I do think `zstd` is supported by Pandas currently: https://pandas.pydata.org/docs/user_guide/io.html?highlight=storage_options#quoting-compression-and-file-format

Reply:
The API hasn't caught up yet: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_parquet.html#pandas.DataFrame.to_parquet

Reply:
yeah, could be the case.

Reply:
Should we do one that's uncompressed?